### PR TITLE
fix Razer Mamba custom effect frames

### PR DESCRIPTION
**Pull request by tildearrow**
https://github.com/terrycain/razer-drivers/issues/215

### CODE REVIEW

**Comment by terricain** [comment]
No. 
https://github.com/terrycain/razer-drivers/blob/master/driver/razermouse_driver.c#L253 
https://github.com/terrycain/razer-drivers/blob/master/driver/razermouse_driver.c#L906
- those lines use 0x0C, dont change that function definition, theres no need.

**Comment by tildearrow** [comment]
What does that mean? I have not touched the previously mentioned lines.

**Comment by terricain** [comment]
Those lines are the area you should be looking in, not the lines you actually touched

**Comment by tildearrow** [comment]
I do not have the DeathAdder Elite.
Yet I'll be looking into it when I get back to my computer.

**Comment by tildearrow** [comment]
Just a question. What do you mean that I should be looking in? Do you mean that I should use razer_chroma_extended_matrix_set_custom_frame instead of the use_0c argument?

**Comment by terricain** [comment]
yeah

**Comment by tildearrow** [comment]
Wait. I think I found a perfect solution that does not require use_0c: razer_chroma_misc_one_row_set_custom_frame. I'm going to use that right now.

**Comment by terricain** [comment]
Ahh yeah technically does the same thing

**Comment by tildearrow** [comment]
Done. If there still is something wrong, please tell me.

File: driver/razercore_driver.c [hunk 1]
```diff
@@ -512,8 +512,9 @@ static int razer_core_probe(struct hid_device *hdev, const struct hid_device_id
     CREATE_DEVICE_FILE(&hdev->dev, &dev_attr_matrix_brightness);
     CREATE_DEVICE_FILE(&hdev->dev, &dev_attr_device_mode);
     
-    if (retval)
+    if (retval) {
```
Comment (z3ntu) [inline driver/razercore_driver.c:515]: Is this necessary...?

Comment (tildearrow) [reply driver/razercore_driver.c:515]: Not really but it seems to get rid of a compiler warning:

```
/home/user0/projects/razer-drivers/driver/razercore_driver.c: In function ‘razer_core_probe’:
/home/user0/projects/razer-drivers/driver/razercore_driver.c:515:5: warning: this ‘if’ clause does not guard... [-Wmisleading-indentation]
     if (retval)
     ^~
/home/user0/projects/razer-drivers/driver/razercore_driver.c:518:2: note: ...this statement, but the latter is misleadingly indented as if it is guarded by the ‘if’
  dev_set_drvdata(&hdev->dev, dev);
  ^~~~~~~~~~~~~~~
```

Comment (terricain) [reply driver/razercore_driver.c:515]: Well if it breaks, there is git blame...
But i see no issue.

**Comment by terricain** [comment]
Seems fine, has it been tested?

File: driver/razercore_driver.c (hunk 1 above)
Comment (z3ntu) [reply driver/razercore_driver.c:515]: @tildearrow Ah then nice 👍

**Comment by tildearrow** [comment]
Under a Razer Mamba, yes, and it works.

However, if talking about the `if (retval) {` bit, then I just thought getting rid of compiler warnings is safer...

**Comment by terricain** [comment]
That's cool, though when I run make I see no warnings. What are you running?

**Comment by tildearrow** [comment]
Under Arch Linux, `make -j8`, via KDevelop.

**Comment by terricain** [comment]
Ahh that's very weird.

Anyway should be fine.

**Comment by z3ntu** [comment]
I could also reproduce the warning at an older revision :)
spectacle p22129

**Pull request merged**